extensions/notebook/src/jupyter/jupyterServerInstallation.ts

```diff
@@ -463,12 +463,16 @@ export class JupyterServerInstallation {
 			}
 
 			if (doUpgrade) {
-				let installPromise = new Promise(async resolve => {
-					if (this._usingConda) {
-						await this.installCondaPackages(condaPackagesToInstall, true);
+				let installPromise = new Promise(async (resolve, reject) => {
+					try {
+						if (this._usingConda) {
+							await this.installCondaPackages(condaPackagesToInstall, true);
+						}
+						await this.installPipPackages(pipPackagesToInstall, true);
+						resolve();
+					} catch (err) {
+						reject(err);
```

Comment:
What's the issue here? Exceptions throw in an async function will result in a rejected promise to begin with so I'm unclear how this is fixing anything.

Reply:
Nevermind, was misreading the promise declaration